### PR TITLE
EZP-28480: SPI cache for URL Handler in 7.0

### PR DESCRIPTION
**Pull request by adamwojs**
> JIRA: https://jira.ez.no/browse/EZP-28480

## Description

SPI cache for URL Handler in 7.0

## TODO

- [x] Remove temporary commit 1d61661a00ca2a7fa56b0eb4178d510c5af9725a before merge. Travis results: https://travis-ci.org/ezsystems/ezpublish-kernel/builds/316841359

### CODE REVIEW

**Comment by andrerom** [comment]
@adamwojs .travis won't run here unless we adjust .travis.yml, can you double check and confirm unit and integration tests passes with this?

File: eZ/Publish/Core/Persistence/Cache/URLHandler.php [hunk 1]
```diff
@@ -110,15 +117,8 @@ public function loadByUrl($url)
...
-
-        $usages = $cache->get();
-        if ($cache->isMiss()) {
-            $this->logger->logCall(__METHOD__, ['url' => $id]);
-            $usages = $this->persistenceHandler->urlHandler()->findUsages($id);
-            $cache->set($usages)->save();
-        }
+        $this->logger->logCall(__METHOD__, ['url' => $id]);
 
-        return $usages;
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Cache/URLHandler.php]: TODO NOTE for later: It should be rather a easy to cache this in 7.0, as you can tag it with content-id and url-id stuff making sure it is cleared on both url cand content changes.


However in 6,x you should probably open PR to remove it, as cache won't be cleared if content is deleted or updated for instance.

File: eZ/Publish/Core/Persistence/Cache/URLHandler.php [hunk 2]
```diff
@@ -19,7 +20,7 @@
 class URLHandler implements URLHandlerInterface
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Cache/URLHandler.php]: Please extend AbstractHandler and remove the duplicated logic

Comment (adamwojs) [reply eZ/Publish/Core/Persistence/Cache/URLHandler.php]: Done 👍

**Comment by adamwojs** [comment]
@andrerom Unit and integration tests passes, but functional tests fails on

> The service "fos_http_cache.user_context.hash_generator" has a dependency o  
  n a non-existent service "ezpublish.user.identity_definer.role_id".          

which seems to be unrealated to this PR. Full travis results: https://travis-ci.org/ezsystems/ezpublish-kernel/builds/316841359

**Comment by andrerom** [comment]
Yes, thats for me to fix on http cache ;)

**Pull request merged**